MIT Press Books.js

```diff
@@ -0,0 +1,298 @@
+{
+	"translatorID": "ac277fbe-000c-46da-b145-fbe799d17eda",
+	"label": "MIT Press Books",
+	"creator": "Guy Aglionby",
+	"target": "https://(www\\.)?mitpress\\.mit\\.edu/(mit-press-open|contributors|search|series|books)",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2019-11-07 18:08:52"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2019 Guy Aglionby
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes('/books/') && !url.includes('/series/')) {
+		return 'book';
+	}
+	else if ((url.includes('/mit-press-open') || url.includes('/contributors/')
+		|| url.includes('/search?') || url.includes('/series/')) && getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === 'multiple') {
+		Zotero.selectItems(getSearchResults(doc, false), function (selected) {
+			if (selected) {
+				ZU.processDocuments(Object.keys(selected), scrape);
+			}
+		});
+	}
+	else {
+		scrape(doc);
+	}
+}
+
+function scrape(doc, url) {
+	let item = new Zotero.Item('book');
+	item.url = url;
+	item.place = 'Cambridge, MA, USA';
+	item.publisher = 'MIT Press';
+	item.language = 'en';
+	item.date = ZU.xpathText(doc, '(//time[@property = "publishDate"]/@content)[1]');
+	item.ISBN = ZU.xpathText(doc, '(//span[@property = "isbn"])[1]');
+	item.numPages = ZU.xpathText(doc, '(//span[@property = "numPages"])[1]');
+	
+	let title = ZU.xpathText(doc, '//h1[@class = "book__title"]').trim();
+	let subtitle = ZU.xpathText(doc, '//h2[@class = "book__subtitle"]').trim();
+	if (subtitle) {
+		item.title = [title, subtitle].join(': ');
```

Comment:
Move the `trim()` from the `subtitle` inside the `if` block, because otherwise there is a potential that you try to trim from `null` which will fail.